### ISSUE TITLE
Crash in AttributionFetcher.trackingProxy.getter, affecting about 3-5% users

### ISSUE THREAD

**Issue by quantamrhino** [RevenueCat/purchases-ios#1694]
- [ yes] I have updated Purchases SDK to the latest version
- [ yes] I have read the [Contribution Guidelines](https://github.com/RevenueCat/purchases-ios/blob/main/Development/CONTRIBUTING.md)
- [yes ] I have searched the [Community](https://community.revenuecat.com)
- [ yes] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [ yes] I have searched for [existing Github issues](https://github.com/RevenueCat/purchases-ios/issues)

**Describe the bug**
Looks like a restore subscription may be being called, and then this crash happens sometimes.

1. Environment
   
    Platform: iOS
    SDK version: 4.5.2
    StoreKit 2 enabled (Y/N): not sure
    OS version: 15.5
    Xcode version: 13
    How widespread is the issue. Percentage of devices affected. 5-7%

2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue - none available
3. Steps to reproduce, with a description of expected vs. actual behavior
 - not always, but somtimes was triggered by a restore Purchases.

5. **Other information** (e.g. stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc.)
```
Crashed: com.apple.root.user-initiated-qos.cooperative
0  libswiftCore.dylib             0x3dcd08 swift_isUniquelyReferenced_nonNull_native + 38
1  MyApp                    0x30fa24 specialized AttributionFetcher.trackingProxy.getter + 4372183588 (<compiler-generated>:4372183588)
2  MyApp                    0x30fdec specialized AttributionFetcher.fetchAuthorizationStatus.getter + 141 (AttributionFetcher.swift:141)
3  MyApp                    0x395bd4 SubscribersAPI.post(receiptData:appUserID:isRestore:productData:presentedOfferingIdentifier:observerMode:subscriberAttributes:completion:) + 4372732884 (<compiler-generated>:4372732884)
4  MyApp                    0x3c0124 closure #2 in closure #1 in PurchasesOrchestrator.syncPurchases(receiptRefreshPolicy:isRestore:completion:) + 98 (Backend.swift:98)
5  MyApp                    0x3c4594 partial apply for closure #2 in closure #1 in PurchasesOrchestrator.syncPurchases(receiptRefreshPolicy:isRestore:completion:) + 4372923796 (<compiler-generated>:4372923796)
6  MyApp                    0x3e5f44 closure #1 in TransactionsManager.customerHasTransactions(receiptData:completion:) + 32 (TransactionsManager.swift:32)
7  libswift_Concurrency.dylib     0x3b7cc swift::runJobInEstablishedExecutorContext(swift::Job*) + 244
8  libswift_Concurrency.dylib     0x3c1e8 swift_job_runImpl(swift::Job*, swift::ExecutorRef) + 72
9  libdispatch.dylib              0x15164 _dispatch_root_queue_drain + 396
10 libdispatch.dylib              0x1596c _dispatch_worker_thread2 + 164
11 libsystem_pthread.dylib        0x1080 _pthread_wqthread + 228
12 libsystem_pthread.dylib        0xe5c start_wqthread + 8
```
**Additional context**

**Comment by NachoSoto** [comment]
Thanks for the report!

This looks the same as #1685.
The fix is in version 4.6.1. Could you test with that?

**Comment by quantamrhino** [comment]
Oh, I see. Looked like a different stack trace, so filed a different one.
I can't repro this locally, only happening in the field. I'll release the update with 4.6.1 and report back. 
Thanks!
